src/app/shared/components/category-navigation/category-navigation.component.ts

```diff
@@ -29,7 +29,7 @@ export class CategoryNavigationComponent {
   constructor(private router: Router) {}
 
   activeRoute(): string | undefined {
-    return this.router.url.split('/')[2];
+    return this.router.url.split('?')[0].split('/')[2];
```

Comment:
zawsze mamy w url '?'

Reply:
może to będzie bardziej kompleksowe:
return !!Object.values(CategoryRoutingName).filter(el => this.router.url.includes(el));

Reply:
If we don't have '?', split returns an array with single value - original string. So this part looks safe.

Your idea makes sense as well. I can change it.